xlr/sdk/src/types.ts

```diff
@@ -9,6 +10,12 @@ export interface ValidationError {
 
   /** Rough categorization of the error type */
   type: "type" | "missing" | "unknown" | "value" | "unexpected";
+
+  /** Expected types */
+  expected?: string[] | string | number | boolean;
```

Comment:
since we'd only be adding `expected` information when we have a `type` error should we make a generic type for validation messages, then specific types for the individual error types, and wrap those up in a union?